bothan-binance/src/worker.rs

```diff
@@ -42,25 +40,22 @@ impl BinanceWorker {
 #[async_trait::async_trait]
 impl AssetWorker for BinanceWorker {
     /// Fetches the AssetStatus for the given cryptocurrency ids.
-    async fn get_assets(&self, ids: &[&str]) -> Vec<AssetStatus> {
+    async fn get_assets(&self, ids: &[&str]) -> Vec<AssetState> {
         self.store.get_assets(ids).await
     }
 
     /// Adds the specified cryptocurrency IDs to the query set and subscribes to their updates.
     async fn add_query_ids(&self, ids: Vec<String>) -> Result<(), Error> {
         let to_sub = self.store.add_query_ids(ids).await;
 
-        if let Err(e) = self.subscribe_tx.send(to_sub.clone()).await {
-            error!("failed to add query ids: {}", e);
-            self.store.remove_query_ids(to_sub.as_slice()).await;
-            Err(Error::ModifyQueryIDsFailed(e.to_string()))
-        } else {
-            Ok(())
-        }
+        self.subscribe_tx
+            .send(to_sub.clone())
+            .await
+            .map_err(|e| Error::ModifyQueryIDsFailed(e.to_string()))
     }
 
     /// Removes the specified cryptocurrency IDs to the query set and subscribes to their updates.
-    async fn remove_query_ids(&self, ids: &[&str]) -> Result<(), Error> {
+    async fn remove_query_ids(&self, ids: Vec<String>) -> Result<(), Error> {
         let to_unsub = self.store.remove_query_ids(ids).await;
 
         if let Err(e) = self.unsubscribe_tx.send(to_unsub.clone()).await {
```

Comment:
do we need to change this function too so it's does not add_query_ids on failure?